src/lib.rs

```diff
@@ -190,209 +224,268 @@ impl KafkaJsonToDelta {
-                                    .await;
-                                WriteAheadLogEntry::new(
-                                    1,
-                                    TransactionState::Prepared,
-                                    None, /*delta table version starts out empty*/
-                                    partition_offsets.clone(),
-                                )
-                            };
-                            wal.put_entry(&wal_entry).await?;
-                            self.log_write_ahead_log_prepared().await;
-
-                            // Complete the delta write
-                            let delta_write_timer = Instant::now();
-                            let txn = action::Action::txn(action::Txn {
-                                appId: self.app_id.clone(),
-                                version: wal_entry.transaction_id,
-                                lastUpdated: std::time::SystemTime::now()
-                                    .duration_since(std::time::UNIX_EPOCH)
-                                    .unwrap()
-                                    .as_millis()
-                                    as i64,
-                            });
-                            self.log_delta_write_started().await;
-                            let commit_result = delta_writer.write_file(&mut vec![txn]).await;
-                            match commit_result {
-                                Ok(version) => {
-                                    self.log_delta_write_completed(&delta_write_timer).await;
-                                    wal.complete_entry(wal_entry.transaction_id, version)
-                                        .await?;
-                                    self.log_write_ahead_log_completed(&wal_timer).await;
-                                }
-                                Err(e) => {
-                                    wal.abort_entry(wal_entry.transaction_id).await?;
-                                    self.log_write_ahead_log_aborted().await;
-                                    self.log_delta_write_failed().await;
-                                    return Err(KafkaJsonToDeltaError::DeltaWriter { source: e });
+        Ok(())
+    }
+
+    async fn run_loop(
+        &mut self,
+        cancellation_token: Option<&CancellationToken>,
+    ) -> Result<(), KafkaJsonToDeltaError> {
+        let mut state = ProcessingState {
+            delta_writer: DeltaWriter::for_table_path(self.table_location.clone()).await?,
+            value_buffers: ValueBuffers::new(),
+            transformer: Transformer::from_transforms(&self.transforms)?,
+            wal: write_ahead_log::new_write_ahead_log(self.app_id.to_string()).await?,
+            latency_timer: Instant::now(),
+        };
+
+        self.init(&state).await?;
+
+        let mut stream = self.consumer.stream();
+        //state.latency_timer = Instant::now(); TODO
```

Comment:
Is this TODO still pending or resolved differently?

Reply:
So that was a leftover of my error debugging above, however let's talk about this one.
Since I moved it from variable to state object, it is now created above, and then reset to now at this place.
Between these 2 calls there's `init` call, so does this init makes a difference so we can have `state.latency_timer = Instant::now` here or rely on the one where struct is created ` latency_timer: Instant::now(),`

Reply:
That's very minor and almost has no impact except the one line of code

Reply:
IMHO - its fine where it is. Allowed latency is going to be on the order of minutes or at least tens of seconds.